### PR TITLE
docs(v1.5.4): add v1.5.4 release notes to CHANGELOG

### PR DESCRIPTION
**Pull request by willvedd**
## Description
Updating changelog for upcoming 1.5.4 release.

~*Note:* This PR is marked as a draft pending #1433 approval and merge.~

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,36 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+
+### Added
+
+* ListUsers API which answers the question "what users are related to a specific object?". This feature is experimental and can be enabled by configuring `OPENFGA_EXPERIMENTALS=enable-list-users`. Also see [Performing a ListUsers call](https://openfga.dev/docs/getting-started/perform-list-users) and [ListUsers API docs](https://openfga.dev/api/service#/Relationship%20Queries/ListUsers). **Known Limitation:** Child usersets that are negated from their parent are currently not returned as `excluded_users` [#1433](https://github.com/openfga/openfga/pull/1433)
+* ListObjects throttling to manage resource usage of expensive queries. Throttling improves overall query performance by limiting the number of dispatches, which are the recursive sub-operations of a ListObjects query [#1571](https://github.com/openfga/openfga/pull/1571)
+* Self-defining usersets for Check, ListObjects and ListUsers. These are implicit tuples that exist by virtue of set theory. For example, the userset `document:1#viewer` implicitly possess the `viewer` relation for `document:1` [#147](https://github.com/openfga/openfga/pull/147)
```
Comment (jon-whit) [inline CHANGELOG.md:19]: This link isn't right I don't think. Did we mean?

https://github.com/openfga/openfga/pull/1066 or
https://github.com/openfga/openfga/pull/1521 ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,35 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+* Stricter filtering of invalid tuples with ListObjects [#1563](https://github.com/openfga/openfga/pull/1563)
+* Panic on server close if caching is enabled [#1568](https://github.com/openfga/openfga/pull/1568)
+
+### Changed
+
+* `request_id` is now same as `trace_id` for easier debugging [#1576](https://github.com/openfga/openfga/pull/1576) - thank you, @00chorch
+
+### Removed
+
+* Deprecated and redundant `craigpastro/openfga-dsl-parse` DSL parser dependency [#1578](https://github.com/openfga/openfga/pull/1578)
```
Comment (miparnisari) [inline CHANGELOG.md:36]: I wouldn't mention this. It doesn't impact end users in any way.

Comment (willvedd) [reply CHANGELOG.md:36]: Is it not a dependency removal? So if OpenFGA is used as a library, its one less dep that gets fetched, no?

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,35 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
 
 ## [Unreleased]
 
+## [1.5.4] - 2024-05-29
```
Comment (miparnisari) [inline CHANGELOG.md:11]: Why aren't none of the performance improvements for Check mentioned?

https://github.com/openfga/openfga/pull/1550
https://github.com/openfga/openfga/pull/1593
https://github.com/openfga/openfga/pull/1589

Comment (willvedd) [reply CHANGELOG.md:11]: Please advise how to articulate

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,35 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
+
+[Full changelog](https://github.com/openfga/openfga/compare/v1.5.3...v1.5.4)
+
+### Added
+
+* ListUsers API which answers the question "what users are related to a specific object?". This feature is experimental and can be enabled by configuring `OPENFGA_EXPERIMENTALS=enable-list-users`. Also see [Performing a ListUsers call](https://openfga.dev/docs/getting-started/perform-list-users) and [ListUsers API docs](https://openfga.dev/api/service#/Relationship%20Queries/ListUsers). **Known Limitation:** Child usersets that are negated from their parent are currently not returned as `excluded_users` [#1433](https://github.com/openfga/openfga/pull/1433)
+* ListObjects throttling to manage resource usage of expensive queries. Throttling improves overall query performance by limiting the number of dispatches, which are the recursive sub-operations of a ListObjects query [#1571](https://github.com/openfga/openfga/pull/1571)
```
Comment (jon-whit) [inline CHANGELOG.md:18]: In addition, do we want to mention https://github.com/openfga/openfga/pull/1546?

File: CHANGELOG.md (hunk 2 above)
Comment (miparnisari) [reply CHANGELOG.md:36]: Yes, but i don't see this as worthy of mentioning. We shuffle dependencies all the time and we never mentioned them in the changelog

Comment (willvedd) [reply CHANGELOG.md:36]: Ok

**Review by jon-whit — DISMISSED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (willvedd) [reply CHANGELOG.md:11]: [#1568](https://github.com/openfga/openfga/pull/1568)

Comment (miparnisari) [reply CHANGELOG.md:11]: ## Changed

- Performance improvements in Check. Less unnecessary spans created (https://github.com/openfga/openfga/pull/1550 and https://github.com/openfga/openfga/pull/1589), less calls to database if the request was cancelled or timed out (https://github.com/openfga/openfga/pull/1593).

Comment (miparnisari) [reply CHANGELOG.md:11]: You dropped #1514 😅 also please mention that is specifically for Check api

Comment (willvedd) [reply CHANGELOG.md:11]: #1514 is an issue. #1593 is already marked as a fix.

**Review by jpadilla — APPROVED** [review]

**Pull request merged**